README.md

```diff
@@ -52,21 +52,30 @@ For example, you can use the following command to trigger ingestion for the list
 where `upload_to_gcs_topic_name` and `gcs_landing_bucket` are the same as the terraform variables of the same name
 
 ## Shared python code
-In order to easily share code between services, most python code should go in the `/python` directory. This directory is set up as a package via the `setup.py` file. Shared code should be put in sub-packages with an `__init__.py` file. If a new sub-package is added, update `setup.py` to specify the name of the sub-package via the `packages` arg.
-
-To work with the code locally, run `pip install ./python` from the root project directory. If your IDE complains about imports after changing code in `/python`, re-run `pip install ./python`.
-
-Note: the `/python` package has three root-level files that aren't necessary: `main.py`, `requirements.in`, and `requirements.txt`. These exist purely so the whole `/python` package can be deployed as a cloud function, in case people are relying on that for development/quick iteration. Due to limitations with cloud functions, these files have to exist directly in the root package. We should eventually remove these.
-
-## Python dependencies
-When developing, if a new dependency is needed:
-1. Add it to the `<service_directory>/requirements.in` file, where `<service_directory>` is the root-level directory for the service that needs the dependency. For example, dependencies needed for `run_ingestion` should go in `run_ingestion/requirements.in`
-2. Run `cd <service_directory>`, then `pip-compile requirements.in`. This will generate a `requirements.txt` file.
-3. Run `pip install -r requirements.txt` to ensure your local environment has the dependencies.
-
-During development you can also just run `pip install <new_dep>`, but remember to add it to `requirements.in` and run `pip-compile requirements.in` before checking in code or deploying the function. Note, you'll need to have followed the python environment setup described above [Python environment setup](#python-environment-setup) to complete these steps.
-
-If dependencies are needed for code that is shared between services, update the `requirements.in` and `requirements.txt` files in each of the services. We may later want to move to a different model, eg one `/python/requirements.in` file for all services, or one per `/python/<sub_package>`. These have tradeoffs like build complexity vs pulling in unnecessary deps, so for now just put deps in each service directly.
+Most python code should go in the `/python` directory, which contains packages that can be installed into any service. Each sub-directory of `/python` is a package with an `__init__.py` file, a `setup.py` file, and a `requirements.in` file. Shared code should go in one of these packages. If a new sub-package is added:
+1. Create a folder `/python/<new_package>`. Inside, add:
+  - An empty `__init__.py` file
+  - A `setup.py` file with options: `name=<new_package>`, `package_dir={'<new_package>': ''}`, and `packages=['<new_package>']`
```

Comment:
is version a required field?

also just curious, will the key for `new_package` in the `package_dir` attribute ever need a value?

Reply:
I thought version was required but I just tried without it and it works fine so I removed it, good catch!

package_dir specifies where to find the source files, so it all depends on the package structure. If your package structure has the files:
```
some_folder
  -> setup.py
  -> ingestion
    -> some_module.py
```
then you wouldn't need to specify package_dir because it matches the package name and it's nested one level under the folder that contains setup.py. Since I didn't see the need to have an additional layer of folder nesting, I put setup.py at the same level as the other files, so then `package_dir={'<new_package>': ''}` tells it that the code is in the current directory (empty string means current directory). This has some good examples for other package layouts: https://docs.python.org/3/distutils/examples.html